src/Agent.Plugins/Artifact/FileContainerProvider.cs

```diff
@@ -191,7 +207,8 @@ private async Task DownloadFileContainerAsync(IEnumerable<FileContainerItem> ite
                             tracer.Info($"Downloading: {targetPath}");
                             if (item.BlobMetadata != null && downloadFromBlob)
                             {
-                                await this.DownloadFileFromBlobAsync(context, containerIdAndRoot, targetPath, projectId, item, dedupClient, clientTelemetry, cancellationToken);
+                                var client = dedupClientTable[GetDomainIdAndDedupIdFromArtifactHash(item.BlobMetadata.ArtifactHash).domainId];
```

Comment:
There's an exception handler building dedupClientTable lookup; could it possibly not be created completely?

Reply:
Any failures to construct the client should result in falling back to downloading from FCS directly because it sets downloadFromBlob = false, so we should not get here.

Reply:
although it looks like the socket exception does not include this logic for some reason.

Reply:
added fallback for the socket exception as well.